polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -1202,6 +1235,64 @@ async fn handle_network_msg<Context>(
 	Ok(())
 }
 
+/// Update collation tracker with the backed and included candidates.
+#[overseer::contextbounds(CollatorProtocol, prefix = crate::overseer)]
+async fn process_block_events<Context>(
+	ctx: &mut Context,
+	collation_tracker: &mut CollationTracker,
+	leaf: Hash,
+	para_id: ParaId,
+	metrics: &Metrics,
+) -> Result<()> {
```

Comment:
another nit: the function returns `Ok(())` in any case. You can drop the return type altoghether.